itensor/tagset.h

```diff
@@ -0,0 +1,228 @@
+//
+// Distributed under the ITensor Library License, Version 1.2
+//    (See accompanying LICENSE file.)
+//
+#ifndef __ITENSOR_TAGSET_H
+#define __ITENSOR_TAGSET_H
+#include "itensor/smallstring.h"
+
+namespace itensor {
+
+class TagSet;
+
+//
+// TagSet
+//
+
+class TagSet
+    {
+    public:
+    using tags_type = std::array<Tag,MAX_TAGS>;
+    private:
+    tags_type tags_;
+    size_t size_ = 0;
+    public:
+
+    TagSet() {};
```

Comment:
```suggestion
    TagSet() {}
```
^ Shouldn't have a semicolon after a method declaration